convert2rhel/actions/handle_packages.py

```diff
@@ -0,0 +1,148 @@
+# Copyright(C) 2023 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import logging
+
+from convert2rhel import actions, pkghandler, pkgmanager, utils
+from convert2rhel.actions.kernel_modules import EnsureKernelModulesCompatibility
+from convert2rhel.actions.special_cases import RemoveIwlax2xxFirmware
+from convert2rhel.actions.subscription import SubscribeSystem
+from convert2rhel.systeminfo import system_info
+
+
+logger = logging.getLogger(__name__)
+
+
+class ListThirdPartyPackages(actions.Action):
+    id = "LIST_THIRD_PARTY_PACKAGES"
+
+    def run(self):
+        """
+        List packages not packaged by the original OS vendor or Red Hat and
+        warn that these are not going to be converted.
+        """
+        logger.task("Convert: List third-party packages")
+
+        third_party_pkgs = pkghandler.get_third_party_pkgs()
+        if third_party_pkgs:
+            logger.warning(
+                "Only packages signed by %s are to be"
+                " replaced. Red Hat support won't be provided"
+                " for the following third party packages:\n" % system_info.name
+            )
+            pkghandler.print_pkg_info(third_party_pkgs)
+            utils.ask_to_continue()
+        else:
+            logger.info("No third party packages installed.")
+
+
+class RemoveExcludedPackages(actions.Action):
+    id = "REMOVE_EXCLUDED_PACKAGES"
+
+    def run(self):
+        """
+        Certain packages need to be removed before the system conversion,
+        depending on the system to be converted.
+        """
+        logger.task("Convert: Remove excluded packages")
+        logger.info("Searching for the following excluded packages:\n")
+        try:
+            pkghandler.remove_pkgs_with_confirm(system_info.excluded_pkgs)
+            # TODO: Handling SystemExit here as way to speedup exception
+            # handling and not refactor contents of the underlying function.
+        except SystemExit as e:
+            # TODO(r0x0d): Places where we raise SystemExit and need to be
+            # changed to something more specific.
+            #   - When we can't remove a package.
+            self.set_result(status="ERROR", error_id="PACKAGE_REMOVAL_FAILED", message=e)
```

Comment:
```suggestion
            self.set_result(status="ERROR", error_id="PACKAGE_REMOVAL_FAILED", message"%s" % e)
```
You may need to change this message to string format since all three params need to be strings

Reply:
Used `str(e)` instead.